logpolicy/logpolicy.go

```diff
@@ -615,16 +615,16 @@ func NewWithConfigPath(collection, dir, cmdName string, netMon *netmon.Monitor)
 	log.SetFlags(0) // other log flags are set on console, not here
 	log.SetOutput(logOutput)
 
-	log.Printf("Program starting: v%v, Go %v: %#v",
+	logf("Program starting: v%v, Go %v: %#v",
```

Comment:
On L615 and L616, we call mutating functions in the "log" package. This seems really weird if we're trying to circumvent the "log" package.

In general, it seems like there should be no calls to the "log" package if a custom `logf` function is provided.

Reply:
That's a good point. Since `func` isn't comparable to anything other than `nil`, I think this means we'll need to special-case `nil` and have that mean to use `log.Printf` and also change global settings. Does that sound right?

Reply:
I think that's fairly normal to check if it's nil and do some default as a result.

Reply:
Okay, did this; PTAL?